### PR TITLE
UI: Zone wizard fix

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR intents to investigate and solve problems with the zone wizard.
So far a problem found is that
- [x] defined vcenters cannot be used in a zone, even though these are not used in any other zone.
- [x] clustername is changed in copmparison to that which the user enters

Fixes: #7563 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [x] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

this has been tested by using the deploy wizard in a vmware environment.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: ui/src/views/infra/zone/ZoneWizardLaunchZone.vue [hunk 1]
```diff
@@ -2048,10 +2053,11 @@ export default {
       })
     },
     addCluster (args) {
+      console.log(args)
```
Comment (DaanHoogland) [inline ui/src/views/infra/zone/ZoneWizardLaunchZone.vue:2056]:
```suggestion
```

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 vmware-67u3

**Review by kiranchavala — APPROVED** [review]
LGTM , Tested manually with 4.18 and vmware 6.7 , the zone creation was successfull

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java [hunk 2]
```diff
@@ -1243,6 +1237,46 @@ public VmwareDatacenterVO addVmwareDatacenter(AddVmwareDcCmd cmd) throws Resourc
...
+        } catch (EntityExistsException e) {
+            // if that fails just get the record as is
+            vmwareDc = vmwareDcDao.getVmwareDatacenterByGuid(guid);
+            // we could now update:
+//            vmwareDc.setPassword(password);
+//            vmwareDc.setUser(user);
+//            vmwareDc.setVmwareDatacenterName(name);
+//            vmwareDc.setVcenterHost(host);
+//            vmwareDcDao.update(vmwareDc.getId(), vmwareDc);
+            // but let's assume user error for now
```
Comment (shwstppr) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java:1254]: Do we need this?
Shouldn't we first check for existing entry and create a new only if doesn't exist

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 3]
```diff
@@ -2510,6 +2508,12 @@ public Boolean doInTransaction(final TransactionStatus status) {
                 for (final VlanVO vlan : vlans) {
                     _vlanDao.remove(vlan.getId());
                 }
+                // we should actually
+//                VmwareDatacenterZoneMapVO vmwareZoneMapping = _vmwareDatacenterZoneMapDao.findByZoneId(zoneId)
+//                if (vmwareZoneMapping != null) {
+//                    _vmwareDatacenterZoneMapDao.removeByZoneId(vmwareZoneMapping.getId());
+//                }
+                // but we don't know about vmware/plugin/hypervisors at this point
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:2516]: Is this needed?

**Review by shwstppr — COMMENTED** [review]
@DaanHoogland is this ready for review/test. I see a couple of commented code blocks.
Added a suggestion for UI file

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java (hunk 2 above)
Comment (DaanHoogland) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java:1254]: my point with this comment is that we could regard the new definition as an update (of more than just the guid)
btw this is not commented out code , but comment written as code. Not sure if that is a justification ;)

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 3 above)
Comment (DaanHoogland) [reply server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:2516]: guess not, just informative
```suggestion
                // we should actually find the mapping and remove if it exists
                // but we don't know about vmware/plugin/hypervisors at this point
```

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java (hunk 2 above)
Comment (DaanHoogland) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java:1254]: what do you think @shwstppr

**Comment by DaanHoogland** [comment]
> @DaanHoogland is this ready for review/test. I see a couple of commented code blocks. Added a suggestion for UI file

yes, it is now @shwstppr ;) thanks for the review/comments

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java (hunk 2 above)
Comment (shwstppr) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java:1254]: I feel a single-line comment would also work. To me, this still feels like commented-out code

**Review by shwstppr — COMMENTED** [review]
Tested UI change while adding an edge zone. LGTM

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java (hunk 2 above)
Comment (DaanHoogland) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java:1254]:
```suggestion
            // we could now update the `vmwareDC` with the user supplied `password`, `user`, `name` and `host`,
            // but let's assume user error for now
```

I must say that I like my original comment better, but how about this, @shwstppr ?

Comment (shwstppr) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java:1254]: fine with me @DaanHoogland

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**